apps/intellij/src/main/kotlin/dev/nx/console/run/NxRunConfigurationProducer.kt

```diff
@@ -0,0 +1,145 @@
+import com.intellij.json.psi.JsonObject
+import com.intellij.json.psi.JsonProperty
+import com.intellij.json.psi.JsonStringLiteral
+import com.intellij.openapi.util.Ref
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.parentOfType
+
+data class NxRunSettings(
+    val nxProjects: String = "",
+    val nxTargets: String = "",
+    val arguments: String = ""
+)
+
+class NxRunConfigurationProducer : LazyRunConfigurationProducer<NxCommandConfiguration>() {
+    override fun getConfigurationFactory(): ConfigurationFactory =
+        NxCommandConfigurationType.getInstance()
+
+    override fun setupConfigurationFromContext(
+        configuration: NxCommandConfiguration,
+        context: ConfigurationContext,
+        sourceElement: Ref<PsiElement>
+    ): Boolean {
+
+        val runSettings =
+            createRunSettingsFromContext(
+                NxRunSettings(
+                    configuration.nxProjects,
+                    configuration.nxTargets,
+                    configuration.arguments
+                ),
+                context,
+                sourceElement
+            )
+
+        if (runSettings == null) {
+            return false
+        } else {
+            setupConfigurationFromSettings(configuration, runSettings)
+            return true
+        }
+    }
+
+    private fun setupConfigurationFromSettings(
+        configuration: NxCommandConfiguration,
+        runSettings: NxRunSettings
+    ) {
+        // configuration.setRunSettings(runSettings)
+        configuration.nxProjects = runSettings.nxProjects
+        configuration.nxTargets = runSettings.nxTargets
+        configuration.arguments = runSettings.arguments
+
+        // configuration.name = "${runSettings.nxProjects}[${runSettings.nxTargets}]"
+        configuration.setGeneratedName()
+    }
+
+    private fun createRunSettingsFromContext(
+        runSettings: NxRunSettings,
+        context: ConfigurationContext,
+        sourceElement: Ref<PsiElement>?
+    ): NxRunSettings? {
+        val element = getElement(context)
+        if (element != null && element.isValid) {
+            val psiProjectJsonFile = getContainingProjectJsonFile(element)
+            if (psiProjectJsonFile == null) {
+                return null
+            } else {
+                val virtualPackageJson = psiProjectJsonFile.virtualFile
+                if (virtualPackageJson == null) {
+                    return null
+                } else {
+                    val targetProperty = findContainingProperty(element)
+                    return if (targetProperty == null) {
+                        null
+                    } else {
+                        val propertyLiteral = element.parent as? JsonStringLiteral
+                        val nxTarget = propertyLiteral?.value ?: return null
+                        val nxProject =
+                            (((propertyLiteral.parent as? JsonProperty)
```

Comment:
While this works, I think we should use the LSP as much as possible to parse JSON for us. Instead of walking up the tree here, you can use the (poorly named) `NxlsService.generatorContextFromPath` method to get information about the current project based on the path of the `project.json` file. 

We should rename that request to be more generic than "generator context" because really all it does is give you the project name and relative path based on a file path. 

@Cammisuli feel free to weigh in on quickly parsing manually vs using the nxls here.